src/neuroconv/datainterfaces/ecephys/axona/axona_utils.py

```diff
@@ -0,0 +1,399 @@
+    current_path = Path(file_path).parent
+
+    path_list = [cur_path.name for cur_path in current_path.rglob("*" + suffix + "*")]
+
+    return path_list
+
+
+def read_all_eeg_file_lfp_data(file_path: FilePathType):
+    """
+    Read LFP data from all Axona `.eeg` or `.egf` files in file_path's directory.
+    E.g. if file_path='/my/directory/my_file.eeg', all .eeg channels will be conactenated
+    to a single np.array (chans x nobs). For .egf files substitude the file suffix.
+    Parameters:
+    -------
+    file_path (Path or Str):
+        Full file_path of Axona `.eeg` or `.egf` file.
+    Returns:
+    -------
+    np.array (chans x obs)
+    """
+
+    file_path_list = get_all_file_paths(file_path)
+    parent_path = Path(file_path).parent
+
+    eeg_memmaps = list()
+    sampling_rates = set()
+    for fname in file_path_list:
+
+        sampling_rates.add(get_eeg_sampling_frequency(parent_path / fname))
+
+        eeg_memmaps.append(read_eeg_file_lfp_data(parent_path / fname))
+    assert len(sampling_rates) < 2, "File headers specify different sampling rates. Cannot combine EEG data."
+
+    eeg_data = np.concatenate(eeg_memmaps, axis=0)
+
+    return eeg_data
+
+
+# Helper functions for AxonaPositionDataInterface
+def parse_generic_header(file_path: FilePathType, params: Union[list, set]):
+    """
+    Given a binary file with phrases and line breaks, enters the
+    first word of a phrase as dictionary key and the following
+    string (without linebreaks) as value. Returns the dictionary.
+
+    INPUT
+    file_path (str): .set file path and name.
+    params (list or set): parameter names to search for.
+
+    OUTPUT
+    header (dict): dictionary with keys being the parameters that
+                   were found & values being strings of the data.
+
+    EXAMPLE
+    parse_generic_header('myset_file.set', ['experimenter', 'trial_time'])
+    """
+    header = dict()
+    if params is not None:
+        params = set(params)
+    with open(file_path, "rb") as f:
+        for bin_line in f:
+            if b"data_start" in bin_line:
+                break
+            line = bin_line.decode("cp1252").replace("\r\n", "").replace("\r", "").strip()
+            parts = line.split(" ")
+            key = parts[0]
+            if params is None or key in params:
+                header[key] = " ".join(parts[1:])
+    return header
+
+
+def read_axona_iso_datetime(set_file: FilePathType):
+    """
+    Creates datetime object (y, m, d, h, m, s) from .set file header
+    and converts it to ISO 8601 format
+    """
+    with open(set_file, "r", encoding="cp1252") as f:
+        for line in f:
+            if line.startswith("trial_date"):
+                date_string = line[len("trial_date") + 1 :].replace("\n", "")
+            if line.startswith("trial_time"):
+                time_string = line[len("trial_time") + 1 :].replace("\n", "")
+    return dateutil.parser.parse(date_string + " " + time_string).isoformat()
```

Comment:
why not just return a datetime object?

Reply:
I think this was done at a time when the metadata did not accept datetime yet. This is just a function that I moved at the request of @CodyCBakerPhD and is only used in the PositionDataInterface that I did not modify. Do you want me to take the opportunity in this PR to change it?

Reply:
I'd say change in subsequent PR - but the original reasoning is correct, this is from before when we had to cast all datetimes to ISO strings in the metadata structure, which is no longer required

Reply:
OK.